src/lxc/conf.c

```diff
@@ -942,6 +942,19 @@ static bool append_ttyname(char **pp, char *name)
 	return true;
 }
 
+static int open_ttymnt_at(int dfd, const char *path)
+{
+	int fd;
+
+	fd = open_at(dfd, path, PROTECT_OPEN | O_CREAT | O_EXCL,
+		     PROTECT_LOOKUP_BENEATH, 0);
+	if (fd < 0 && (errno == ENXIO || errno == EEXIST))
+		fd = open_at(dfd, path, PROTECT_OPATH_FILE,
+			     PROTECT_LOOKUP_BENEATH, 0);
```

Comment:
Hm, I'd like to understand when this happens. So, if we used `lxc.autodev = 1` then we've mounted a new `tmpfs` instance over `/dev` which means the device files can't possible exist before. Can you outline your usecase?

Reply:
In my scenario devtmpfs is mounted on /dev, because containers are used to run multiple instances of SW which has been designed to exist only in one instance on the system and creating per container dev would make it more complex. The host distribution creates /dev/tty1 and as there is just one devtmpfs instance in Linux, it exists in the container as well and because the underlying device is not there, LXC fails to open it and stops.

This used to work in the past as this issue popped up after updating to LXC4 without changing anything else.

Reply:
Ah, ok thank you.